World.py

```diff
@@ -60,17 +60,17 @@ def __init__(self, id, settings, resolveRandomizedSettings=True):
         self.entrance_shuffle = (
             self.shuffle_interior_entrances or settings.shuffle_grotto_entrances or self.shuffle_dungeon_entrances
             or settings.shuffle_overworld_entrances or settings.owl_drops or settings.warp_songs
-            or settings.spawn_positions or (settings.shuffle_bosses != 'off')
+            or (settings.spawn_positions != 'off') or (settings.shuffle_bosses != 'off')
         )
 
-        self.ensure_tod_access = self.shuffle_interior_entrances or settings.shuffle_overworld_entrances or settings.spawn_positions
+        self.ensure_tod_access = self.shuffle_interior_entrances or settings.shuffle_overworld_entrances or (settings.spawn_positions != 'off')
         self.disable_trade_revert = self.shuffle_interior_entrances or settings.shuffle_overworld_entrances
 
         if (
             settings.open_forest == 'closed'
             and (
                 self.shuffle_special_interior_entrances or settings.shuffle_overworld_entrances
-                or settings.warp_songs or settings.spawn_positions or (settings.shuffle_bosses != 'off')
+                or settings.warp_songs or (settings.spawn_positions != 'off') or (settings.shuffle_bosses != 'off')
```

Comment:
Shuffling adult spawn doesn't break Closed Forest.

```suggestion
                or settings.warp_songs or settings.spawn_positions in ('child', 'both') or (settings.shuffle_bosses != 'off')
```

Reply:
Why was this reverted?